### PR TITLE
Feat/chat send message

### PR DESCRIPTION
**Pull request by mthinh**
# What ?
- Implement sending messages
- Implement mention action

# Screenshot 
Screenshot 2023-08-04 at 10 58 20

### CODE REVIEW

File: lib/core/presentation/pages/chat/chat_message/widgets/chat_input/chat_input_widget.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          child: Padding(
+            padding: const EdgeInsets.symmetric(vertical: 4.0),
+            child: InputBar(
+              room: controller.room,
+              minLines: 1,
+              maxLines: 8,
+              autofocus: false,
+              keyboardType: TextInputType.multiline,
+              onSubmitted: controller.onInputBarSubmitted,
+              // onSubmitImage: controller.sendImageFromClipBoard,
```
Comment (sieuhuflit) [inline lib/core/presentation/pages/chat/chat_message/widgets/chat_input/chat_input_widget.dart:36]: add TODO note, better to trace

File: lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_suggestion_box_widget.dart [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      );
+    }
+    return const SizedBox.shrink();
+  }
+}
+
+String commandHint(String command) {
+  switch (command) {
+    case "ban":
+      return "Ban";
```
Comment (sieuhuflit) [inline lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_suggestion_box_widget.dart:133]: Do we need should define enum, or this is for localization display ?

File: lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_widget.dart [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+            'displayname': user.displayName,
+            'avatar_url': user.avatarUrl?.toString(),
+          });
+        }
+        if (ret.length > maxResults) {
+          break;
+        }
+      }
+    }
+    final roomMatch = RegExp(r'(?:\s|^)#([-\w]+)$').firstMatch(searchText);
```
Comment (sieuhuflit) [inline lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_widget.dart:165]: Recommend define this regex pattern somewhere , like pattern constants 

And we define good naming regex there

Example: mention_regex, naming_regex ...etc..

File: lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_widget.dart [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+  void insertSuggestion(_, Map<String, String?> suggestion) {
+    final replaceText = controller!.text.substring(0, controller!.selection.baseOffset);
+    var startText = '';
+    final afterText =
+        replaceText == controller!.text ? '' : controller!.text.substring(controller!.selection.baseOffset + 1);
+    var insertText = '';
+    if (suggestion['type'] == 'command') {
+      insertText = '${suggestion['name']!} ';
+      startText = replaceText.replaceAllMapped(
+        RegExp(r'^(/\w*)$'),
```
Comment (sieuhuflit) [inline lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_widget.dart:205]: same regex

File: lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_widget.dart [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+            break;
+          }
+        }
+        if (!isUnique) {
+          break;
+        }
+      }
+      insertText = ':${isUnique ? '' : '${insertPack!}~'}$insertEmote: ';
+      startText = replaceText.replaceAllMapped(
+        RegExp(r'(\s|^)(:(?:[-\w]+~)?[-\w]+)$'),
```
Comment (sieuhuflit) [inline lib/core/presentation/pages/chat/chat_message/widgets/chat_input/input_bar_widget.dart:237]: Regex

**Comment by mthinh** [comment]
@sieuhuflit resolved all. Please help recheck

**Review by sieuhuflit — APPROVED** [review]
LGTM 💯

**Pull request merged**